rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -819,10 +819,85 @@ void getFederationElection_whenElectionIsNotInStorage_shouldReturnDefault() {
         // Assert
 
         byte[] expectedElectionEncoded = BridgeSerializationUtils.serializeElection(expectedElection);
+
         assertArrayEquals(expectedElectionEncoded, serializeElection(actualElection));
 
     }
 
+    @Test
+    void getActiveFederationCreationBlockHeight_beforeRSKIP186_storageIsNotAccessedAndReturnsEmpty() {
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(false);
+
+        // Arrange
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        // Putting some value in the storage just to then assert that before fork, the storage won't be accessed.
+        storageAccessor.saveToRepository(ACTIVE_FEDERATION_CREATION_BLOCK_HEIGHT_KEY.getKey(), new byte[] { 1 });
+
+        // Act
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Assert
+        assertEquals(Optional.empty(), federationStorageProvider.getActiveFederationCreationBlockHeight(activations));
+
+    }
+
+    @Test
+    void getActiveFederationCreationBlockHeight_afterRSKIP186_getsValueFromStorage() {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP186)).thenReturn(true);
+
+        // Arrange
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        long expectedValue = 1;
```

Comment:
nit, but I would use a larger value to make it a bit more realistic. Just in case, we had some issues in the past by using values lower than the ones used in mainnet.

Anything above 1,000,000 is probably alright

Reply:
Updated.